### PR TITLE
Add code coverage reporting

### PR DESCRIPTION
**Pull request by odsod**
- docs: add badges to README.md
- ci: add GitHub Actions to dependabot.yml
- ci: add code coverage reporting

### CODE REVIEW

File: .github/codecov.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+codecov:
+  require_ci_to_pass: yes
+
+ignore:
+  - internal/examples/proto/gen
+
+coverage:
+  range: "70...100"
```
Comment (Jassob) [inline .github/codecov.yml:8]: Does this means that we require all packages to have at least 70% coverage and otherwise the coverage check fails?

**Review by Jassob — COMMENTED** [review]
Nice! :100: 

Only one question:

File: .github/codecov.yml (hunk 1 above)
Comment (odsod) [reply .github/codecov.yml:8]: It should just show up as red for now.

**Review by Jassob — APPROVED** [review]
:tada:

**Pull request merged**